core/src/main/java/org/infinispan/persistence/spi/CacheWriter.java

```diff
@@ -35,4 +38,28 @@
     * @throws PersistenceException in case of an error, e.g. communicating with the external storage
     */
    boolean delete(Object key);
+
+   /**
+    * Persist all provided entries to the store in a single batch update. If this is not supported by the
+    * underlying store, then entries are written to the store sequentially via {@link #write(MarshalledEntry)}
+    *
+    * @param entries a List of MarshalledEntry to be written to the store.
+    * @throws NullPointerException if entries is null.
+    */
+   default void writeBatch(List<MarshalledEntry<? extends K, ? extends V>> entries) {
+      Objects.requireNonNull(entries);
+      entries.forEach(this::write);
+   }
+
+   /**
+    * Remove all provided keys from the store in a single batch operation. If this is not supported by the
+    * underlying store, then keys are removed from the store sequentially via {@link #delete(Object)}
+    *
+    * @param keys a List of MarshalledEntry to be removed from the store.
+    * @throws NullPointerException if keys is null.
+    */
+   default void deleteBatch(List<Object> keys) {
```

Comment:
I am thinking we want to implement this with `SingleFileStore` as well. In this case it can first order the entries to remove so that the disk updates are done sequentially to more efficiently write to disk. Take a look at the `process` method which does something similar.